bagelbids/cli.py

```diff
@@ -15,6 +15,35 @@
 DICTIONARY_SCHEMA = dictionary_models.DataDictionary.schema()
 
 
+def generate_context():
+    # Direct copy of the dandi-schema context generation function
+    # https://github.com/dandi/dandi-schema/blob/c616d87eaae8869770df0cb5405c24afdb9db096/dandischema/metadata.py
+    import pydantic
+
+    field_preamble = {
+        "bagel": "http://neurobagel.org/vocab/",
+        "snomed": "https://identifiers.org/snomedct:",
+        "nidm": "http://purl.org/nidash/nidm#",
+    }
+    fields = {}
+    for val in dir(models):
+        klass = getattr(models, val)
+        if not isinstance(klass, pydantic.main.ModelMetaclass):
+            continue
+        fields[klass.__name__] = "bagel:" + klass.__name__
```

Comment:
I think there are two questions here:

> why we couldn't use val

`val` is a string name in the `models` namespace that refers to something. we get the "something" that `val` refers to with `getattr` and call it `klass`. if `klass` is a class (e.g. our pydantic ones in `models`),  then it will have a `__name__` attribute that is the string version of its class name. But that doesn't have to be the same thing as `val`.

For example in `models` you could just say:
```python
class Dataset(Bagel):
  pass

MyFunnyReference = Dataset # because we don't use (), this is a reference to the class, not an instance of it
```

Now `val` will be `MyFunnyReference`, but `klass.__name__` will be `Dataset`. 

> why we couldn't use val here for simplicity

I don't think it would be simpler / more readable. The entire section here is concerned with getting pydantic classes in the `models` namespace and then unwrapping their contents into the `fields` dictionary that will become our context. Even if we're sure that `val == klass.__name__` you would still have to remember that `dir` returns string references of things to you and that `val` is probably the name of a class we are interested in. But you'd still have to stop and look what `val` was to be sure:

```python
klass = getattr(models, val)
if not isinstance(klass, pydantic.main.ModelMetaclass):
    continue
fields[val] = "bagel:" + val
for name, field in klass.__fields__.items():
```

Does not read more clearly to me.

Reply:
Okay, that's fair!